users/service.go

```diff
@@ -28,76 +28,85 @@ var (
 	return service{
 		token:      token,
-		clients:    crepo,
+		users:      urepo,
 		policies:   policyService,
 		hasher:     hasher,
 		email:      emailer,
 		idProvider: idp,
 	}
 }
 
-func (svc service) RegisterClient(ctx context.Context, session authn.Session, cli mgclients.Client, selfRegister bool) (rc mgclients.Client, err error) {
+func (svc service) Register(ctx context.Context, session authn.Session, u User, selfRegister bool) (uc User, err error) {
 	if !selfRegister {
 		if err := svc.checkSuperAdmin(ctx, session); err != nil {
-			return mgclients.Client{}, err
+			return User{}, err
 		}
 	}
 
-	clientID, err := svc.idProvider.ID()
+	userID, err := svc.idProvider.ID()
 	if err != nil {
-		return mgclients.Client{}, err
+		return User{}, err
 	}
 
-	if cli.Credentials.Secret != "" {
-		hash, err := svc.hasher.Hash(cli.Credentials.Secret)
+	if u.Credentials.Secret != "" {
+		hash, err := svc.hasher.Hash(u.Credentials.Secret)
 		if err != nil {
-			return mgclients.Client{}, errors.Wrap(svcerr.ErrMalformedEntity, err)
+			return User{}, errors.Wrap(svcerr.ErrMalformedEntity, err)
 		}
-		cli.Credentials.Secret = hash
+		u.Credentials.Secret = hash
 	}
 
-	if cli.Status != mgclients.DisabledStatus && cli.Status != mgclients.EnabledStatus {
-		return mgclients.Client{}, errors.Wrap(svcerr.ErrMalformedEntity, svcerr.ErrInvalidStatus)
+	if u.Status != DisabledStatus && u.Status != EnabledStatus {
+		return User{}, errors.Wrap(svcerr.ErrMalformedEntity, svcerr.ErrInvalidStatus)
 	}
-	if cli.Role != mgclients.UserRole && cli.Role != mgclients.AdminRole {
-		return mgclients.Client{}, errors.Wrap(svcerr.ErrMalformedEntity, svcerr.ErrInvalidRole)
+	if u.Role != UserRole && u.Role != AdminRole {
+		return User{}, errors.Wrap(svcerr.ErrMalformedEntity, svcerr.ErrInvalidRole)
 	}
-	cli.ID = clientID
-	cli.CreatedAt = time.Now()
+	u.ID = userID
+	u.CreatedAt = time.Now()
 
-	if err := svc.addClientPolicy(ctx, cli.ID, cli.Role); err != nil {
-		return mgclients.Client{}, err
+	if err := svc.addUserPolicy(ctx, u.ID, u.Role); err != nil {
+		return User{}, err
 	}
 	defer func() {
 		if err != nil {
-			if errRollback := svc.addClientPolicyRollback(ctx, cli.ID, cli.Role); errRollback != nil {
+			if errRollback := svc.addUserPolicyRollback(ctx, u.ID, u.Role); errRollback != nil {
 				err = errors.Wrap(errors.Wrap(errors.ErrRollbackTx, errRollback), err)
 			}
 		}
 	}()
-	client, err := svc.clients.Save(ctx, cli)
+	user, err := svc.users.Save(ctx, u)
 	if err != nil {
-		return mgclients.Client{}, errors.Wrap(svcerr.ErrCreateEntity, err)
+		return User{}, errors.Wrap(svcerr.ErrCreateEntity, err)
 	}
-	return client, nil
+	return user, nil
 }
 
-func (svc service) IssueToken(ctx context.Context, identity, secret string) (*magistrala.Token, error) {
-	dbUser, err := svc.clients.RetrieveByIdentity(ctx, identity)
+func (svc service) IssueToken(ctx context.Context, email, username, secret string) (*magistrala.Token, error) {
```

Comment:
Since we need only one identity (username or email), let's not have one more parameter than we need. We can check if the identity is email or username inside the method.